src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -171,7 +172,10 @@ namespace ts.codefix {
 
             const properties = arrayFrom(checker.getUnmatchedProperties(checker.getTypeAtLocation(parent), checker.getTypeAtLocation(param), /* requireOptionalProperties */ false, /* matchDiscriminantProperties */ false));
             if (!length(properties)) return undefined;
-            return { kind: InfoKind.ObjectLiteral, token: param.name, properties, indentation: 0, parentDeclaration: parent };
+
+            const formatOptions = getFormatCodeSettingsForWriting(formatContext, sourceFile);
+            const indentation = formatting.SmartIndenter.getIndentation(getLineStartPositionForPosition(tokenPos, sourceFile), sourceFile, formatOptions, getLineStartPositionForPosition(tokenPos, sourceFile) === tokenPos);
+            return { kind: InfoKind.ObjectLiteral, token: param.name, properties, indentation, trimLeadingWhiteSpaces: true, parentDeclaration: parent };

```

Comment:
The next non-undefined return in this function a few lines down has `indentation: undefined`. Does that one need to be updated too? Or, what happens if you just remove `indentation: 0` here instead of explicitly setting it? I don’t immediately see a reason why these two returns should be so different when it comes to indentation. I would suspect that either both are broken and both need the fix you applied here, or that just returning an `undefined` indentation is enough to fix this one. But I’m not sure without running the code.

Reply:
If we remove `indendation: 0` -> `indentation: undefined` which leads to the scenario below
```js
f(() => {
    g({
        foo: "",
        bar: function(a: string): void {
            throw new Error("Function not implemented.");
        }
    });
});
```
- in this function (which is called later for applying changes) https://github.com/microsoft/TypeScript/blob/main/src/services/textChanges.ts#L1082, indentation will be assigned to 12 (8 from https://github.com/microsoft/TypeScript/blob/main/src/services/formatting/smartIndenter.ts#L73 and 4 from https://github.com/microsoft/TypeScript/blob/main/src/services/textChanges.ts#L1096).

For the line below, I cannot find test cases to double check so I haven't touched them yet (not sure should I touch it in this pr or another pr).
```js
       if (isIdentifier(token) && hasInitializer(parent) && parent.initializer && isObjectLiteralExpression(parent.initializer)) {
            const properties = arrayFrom(checker.getUnmatchedProperties(checker.getTypeAtLocation(parent.initializer), checker.getTypeAtLocation(token), /* requireOptionalProperties */ false, /* matchDiscriminantProperties */ false));
            if (!length(properties)) return undefined;
            return { kind: InfoKind.ObjectLiteral, token, properties, indentation: undefined, parentDeclaration: parent.initializer };
        }
```

Reply:
I think the best change would be to fix the formatter/textChanges logic that causes this indentation to be off. Codefixes generally shouldn’t need to worry about these things in their own logic. It seems like the formatter should be capable of figuring out the right indentation here. That will probably be a more difficult bug to figure out though 😬

Reply:
I would say so if it is possible (it might need a lot of changes and could be difficult 🙈). What do you think? Should we merge this one to fix the issue first (we have a test case for this bug to make sure it won't happen again) and later to I will dig into formatter/textChanges (or someone else).

Reply:
I think I’m ok with this targeted fix for now, but it would be nice to know if we need the same fix a few lines down. You could try replacing `indentation: undefined` with something like `indentation: 42` or something definitely wrong in order to find tests that touch that code.

Reply:
hi @andrewbranch, thank for your thought. After digging, I think it makes more sense to set a fixed indentation in fixing object literal file. When applying textChanges  https://github.com/microsoft/TypeScript/blob/main/src/services/textChanges.ts#L1078, it does
- Format input code without indentation.
- Indent the output code above.
In this case, it is a little special, we want to treat like block. if `{` starts in any position (can be in the middle of line), the following indentation should treat `{` as starting of that line (including leading whitespace).
```
    const a: { x: undefined, y: undefined } = {}       // leading 4 whitespaces and { starts in the middle of line
->
    const a: { x: undefined, y: undefined } = {
        x: undefined,
        y: undefined,
    }
---------------------
    const a: {x : undefined, y: undefined } =
     {}
->
    const a: { x: undefined, y: undefined } =
     {                                                  // leading 5 whitespaces and { starts at 6 column
         x: undefined,
         y: undefined,
     }
```